src/aks-preview/azext_aks_preview/aks_app_commands/aksappcommands.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+import shutil
+import subprocess
+import requests
+import os
+import platform
+from pathlib import Path
+
+
+def aks_web_app_init():
+    filePath = binaryPreCheck()
+    if filePath:
```

Comment:
💡 How about:

We structure it like this, just so that we have single responsibility key functs call are at the top level function and we can have conditional checks.

pseudo code drafts idea.

``` *.py

    filePath = binaryPreCheck() # returns path or nothing
    
    if filePath is nothing:
         # download is not required move on.
         isRunSuccessful = run_binary(filePath)

   # if filePath / binary doesn't exist then download and move on. 
   if isRunSuccessful:
         cmd_finish()
             return
```